exonum-java-binding-core/src/test/java/com/exonum/binding/blockchain/BlockchainIntegrationTest.java

```diff
@@ -0,0 +1,293 @@
+        throws CloseFailuresException {
+      try (Cleaner cleaner = new Cleaner()) {
+        List<Block> blocks = LongStream.rangeClosed(0, blockchainHeight)
+            .mapToObj(h -> aBlock(h).build())
+            .map(Blocks::withProperHash)
+            .collect(Collectors.toList());
+
+        // Add the blocks
+        // … to the map of blocks
+        Fork fork = database.createFork(cleaner);
+        MapIndex<HashCode, Block> blocksByHash = MapIndexProxy.newInstance(TEST_BLOCKS, fork,
+            StandardSerializers.hash(), BlockSerializer.INSTANCE);
+
+        for (Block b : blocks) {
+          blocksByHash.put(b.getBlockHash(), b);
+        }
+
+        // … to the list of block hashes
+        ListIndex<HashCode> blockHashes = ListIndexProxy.newInstance(TEST_BLOCK_HASHES, fork,
+            StandardSerializers.hash());
+        for (Block b : blocks) {
+          blockHashes.add(b.getBlockHash());
+        }
+
+        // Merge the changes
+        database.merge(fork);
+
+        return blocks;
+      }
+    }
+
+    private void setupMocks(Snapshot snapshot) {
+      when(coreSchema.getHeight()).thenReturn(HEIGHT);
+      MapIndex<HashCode, Block> blocksByHash = MapIndexProxy.newInstance(TEST_BLOCKS, snapshot,
+          StandardSerializers.hash(), BlockSerializer.INSTANCE);
+      lenient().when(coreSchema.getBlocks()).thenReturn(blocksByHash);
+      ListIndex<HashCode> blockHashes = ListIndexProxy.newInstance(TEST_BLOCK_HASHES, snapshot,
+          StandardSerializers.hash());
+      lenient().when(coreSchema.getBlockHashes()).thenReturn(blockHashes);
+    }
+
+    @Test
+    void getBlockAtHeight() {
+      long blockchainHeight = HEIGHT;
+
+      // Test that correct blocks are returned
+      for (int height = 0; height <= blockchainHeight; height++) {
+        Block actualBlock = blockchain.getBlock(height);
+        Block expectedBlock = blocks.get(height);
+
+        assertThat(actualBlock).isEqualTo(expectedBlock);
+      }
+      // Test no blocks beyond the height
+      assertThrows(IndexOutOfBoundsException.class,
+          () -> blockchain.getBlock(blockchainHeight + 2));
+    }
+
+    @ParameterizedTest
+    @ValueSource(longs = {
+        Long.MIN_VALUE, -1, HEIGHT + 1, HEIGHT + 2, Long.MAX_VALUE
+    })
+    void getBlockAtInvalidHeight(long invalidBlockchainHeight) {
+      Exception e = assertThrows(IndexOutOfBoundsException.class,
+          () -> blockchain.getBlock(invalidBlockchainHeight));
+
+      assertThat(e).hasMessageContaining(Long.toString(invalidBlockchainHeight));
+    }
+  }
+
+  /**
+   * Creates a builder of a block, fully initialized with defaults, inferred from the height.
+   * An invocation with the same height will produce exactly the same builder. The block hash
+   * is <strong>not</strong> equal to the hash of the block with such parameters.
+   *
+   * @param blockHeight a block height
+   * @return a new block builder
+   */
+  private static Builder aBlock(long blockHeight) {
```

Comment:
what is the reason for having such method name? And configure checkstyle for it. How about `newBlock`?

Reply:
It's a kind of a commonly used pattern for naming methods in test classes that return a preconfigured builder with some defaults (hence 'a' = 'some', 'one about properties of which I do not care in this case') that are overridden in tests that need to set some properties.